lib/src/util/leak_detector_visitor.dart

```diff
@@ -126,6 +128,21 @@ bool _hasMatch(Map<DartTypePredicate, String> predicates, DartType type,
         (bool previous, DartTypePredicate p) =>
             previous || p(type) && predicates[p] == methodName);
 
+bool _isInvocationThroughCascadeExpression(
+        MethodInvocation invocation, VariableDeclaration variable) {
+  if (invocation.realTarget is! SimpleIdentifier) {
+    return false;
+  }
+
+  final identifier = invocation.realTarget as SimpleIdentifier;
+  if (identifier.bestElement is! PropertyAccessorElement) {
+    return false;
+  }
+
+  final element = identifier.bestElement as PropertyAccessorElement;
+  return element.variable == variable.element;
+}
```

Comment:
Just something to consider: There is a slightly different coding style that would result in shorter code (fewer lines, fewer casts):
```
final identifier = invocation.realTarget;
if (identifier is SimpleIdentifier) {
  final element = identifier.bestElement;
  if (element is PropertyAccessorElement) {
    return element.variable == variable.element;
  }
}
return false;
```